addons/pkg-npm/model-java/src/main/java/org/commonjava/indy/pkg/npm/model/PackageMetadata.java

```diff
@@ -411,92 +374,87 @@ public boolean merge( PackageMetadata source )
 
         // merge time map
         SimpleDateFormat sdf = new SimpleDateFormat( TIMEOUT_FORMAT );
-        Map<String, String> sourceTimes = source.getTime();
+        Map<String, Date> clone = new LinkedHashMap<>();
+
+        for ( String key : time.keySet() )
+        {
+            String value = time.get( key );
+            Date date = null;
+            try
+            {
+                date = sdf.parse( value );
+            }
+            catch ( ParseException e )
+            {
+                logger.error( String.format( "Cannot parse date: %s. Reason: %s", value, e ) );
+            }
+            clone.put( key, date );
+        }
 
-        Iterator timeKey = sourceTimes.keySet().iterator();
+        Map<String, String> sourceTimes = source.getTime();
+        Iterator versionName = sourceTimes.keySet().iterator();
         boolean added = false;
 
-        while ( timeKey.hasNext() )
+        while ( versionName.hasNext() )
         {
-            String v = (String) timeKey.next();
-            String value = sourceTimes.get( v );
-            boolean s = false;
-            Iterator snapshot = time.keySet().iterator();
-
-            while ( snapshot.hasNext() )
+            String key = (String) versionName.next();
+            String value = sourceTimes.get( key );
+            Date date = null;
+            try
             {
-                String preExisting = (String) snapshot.next();
-                // when the source's version update time is more recent, will update it into the original map.
-                if ( preExisting.equals( v ) )
+                date = sdf.parse( value );
+            }
+            catch ( ParseException e )
+            {
+                logger.error( String.format( "Cannot parse date: %s. Reason: %s", value, e ) );
+            }
+            if ( clone.keySet().contains( key ) )
```

Comment:
`clone.containsKey()` ? It might be a cheaper call (I'm not sure how much work goes into rendering a set for keySet()).